reactjs-films-homework/src/components/Header/Form/__tests__/Form.spec.tsx

```diff
@@ -1,20 +1,34 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, cleanup } from '@testing-library/react';
 import Form from '../Form';
+import { Provider } from 'react-redux';
+import store from '../../../../store/store';
 
 describe('Form', () => {
-  render(<Form />);
-  it('Form snapshot', () => {
-    const { asFragment } = render(<Form />);
-    //   screen.debug();
-    expect(asFragment).toMatchSnapshot();
+  let fragment: any;
+
+  beforeEach(() => {
+    const { asFragment } = render(
+      <Provider store={store}>
+        <Form placeholder="Movies" />
+      </Provider>
+    );
+    fragment = asFragment();
+  });
+
+  afterEach(() => {
+    cleanup();
   });
-  it('placeholder text', () => {
-    render(<Form />);
+
+  it('Snapshot', () => {
+    expect(fragment).toMatchSnapshot();
+  });
+
+  it('Should render placeholder', () => {
     expect(screen.getByPlaceholderText(/Movies/i)).toBeInTheDocument();
   });
+
   it('Must has not default value', () => {
-    render(<Form />);
     expect(screen.getByDisplayValue('')).toBeInTheDocument();
   });
```

Comment:
There are some conventions how your tests should be structured.
Consider the AAA pattern:

AAA (Arrange-Act-Assert) pattern
The AAA (Arrange-Act-Assert) pattern has become almost a standard across the industry. It suggests that you should divide your test method into three sections: arrange, act and assert. Each one of them only responsible for the part in which they are named after.

In Arrange section you only have code required to setup that specific test. Here objects would be created, mocks setup (if you are using one) and potentially expectations would be set.

Then there is the Act, which should be the invocation of the method being tested.

And on Assert you would simply check whether the expectations were met.

Reply:
done, I rewrote all tests